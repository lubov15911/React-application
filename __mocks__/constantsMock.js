export const moviePreviewData = {
    poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
    title: 'Coco',
    vote_average: 7.8,
    tagline: 'The celebration of a lifetime',
    release_date: '2017-10-27',
    runtime: 105,
    genres: ["Adventure", "Comedy", "Family", "Animation"],
    overview: 'Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz.',
};

export const filmsData = [{
    id: 354912,
    poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
    title: 'Coco',
    release_date: '2017-10-27',
    genres: ["Adventure", "Comedy", "Family", "Animation"],
}, {
    id: 269149,
    poster_path: "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
    title: "Zootopia",
    release_date: "2016-02-11",
    genres: ["Animation", "Adventure", "Family", "Comedy"],
}, {
    id: 333339,
    poster_path: 'https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg',
    title: 'Ready Player One',
    release_date: '2018-03-28',
    genres: ["Adventure", "Science Fiction", "Action"],
}, {
    id: 337167,
    poster_path: "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
    title: "Fifty Shades Freed",
    release_date: "2018-03-28",
    genres: ["Drama", "Romance"],
}];

export const searchOptionsData = {
    title: 'Title',
    genre: 'Genre'
};

export const sortOptionsData = {
    rating: 'Rating',
    release: 'Release date'
};

export const searchValueData = 'Ready';
