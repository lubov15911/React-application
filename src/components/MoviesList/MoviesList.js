import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import MoviesListItem from '../MoviesListItem';

import './MoviesList.scss'

const propTypes = {
    films: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const MoviesList = ({ films }) => (
    <ul className="movies-list">
        {films.map(item => {
            return <MoviesListItem movieData={item} key={item.id} />;
        })}
    </ul>
);
MoviesList.propTypes = propTypes;

const mapStateToProps = (state) => {
    return {
        films: state.films,
    }
};
export default connect(mapStateToProps)(MoviesList);
