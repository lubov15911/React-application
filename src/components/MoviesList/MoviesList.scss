.movies-list {
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-column-gap: 6em;
  list-style: none;
  margin: 0;
  padding: 0;
}
