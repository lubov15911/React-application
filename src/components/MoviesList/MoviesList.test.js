import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import MoviesList from './MoviesList';

import { filmsData } from '../../../__mocks__/constantsMock';

describe('MoviesList', () => {
    it('should render correctly', () => {
        const mockStore = configureStore()({
            films: filmsData,
        });

        const component = shallow(<MoviesList store={mockStore} />).dive();

        expect(component).toMatchSnapshot();
    });
});
