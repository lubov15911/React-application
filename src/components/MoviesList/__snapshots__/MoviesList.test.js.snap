// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should render correctly 1`] = `
<ul
  className="movies-list"
>
  <MoviesListItem
    key="354912"
    movieData={
      Object {
        "genres": Array [
          "Adventure",
          "Comedy",
          "Family",
          "Animation",
        ],
        "id": 354912,
        "poster_path": "https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
        "release_date": "2017-10-27",
        "title": "Coco",
      }
    }
    raiseClickEvent={[Function]}
  />
  <MoviesListItem
    key="333339"
    movieData={
      Object {
        "genres": Array [
          "Adventure",
          "Science Fiction",
          "Action",
        ],
        "id": 333339,
        "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
        "release_date": "2018-03-28",
        "title": "Ready Player One",
      }
    }
    raiseClickEvent={[Function]}
  />
</ul>
`;
